python-exercises/assignment/parking_lot.py

```diff
@@ -0,0 +1,239 @@
+                    self.car_count += 1
+                    break
+            return 'OK'
+
+    def leave(self, slot):
+        """
+        This function is to leave a car in current slot
+        """
+        if self.car_count == 0:
+            return 'No car to leave!'
+        elif slot > self.size or slot < 0:
+            return 'No slot in this parking lot'
+        else:
+            if self.parking_lot_list[slot - 1].car is not None:
+                self.parking_lot_list[slot - 1].car = None
+                self.car_count -= 1
+                return 'OK'
+            else:
+                return 'No car to leave in this slot!'
+    
+    def get_status(self):
+        """
+        This function is to get the current status of the parking lot
+        """
+        self.table.clear()
+        self.table.field_names = ['Slot No.', 'Reg No.', 'Color']
+        for slot in self.parking_lot_list:
+            if slot.car is None:
+                continue
+            else:
+                self.table.add_rows([[str(slot.id), str(slot.get_registration_number()), str(slot.get_color())]])
+        print(self.table)
+        return 'OK'
+    
+    def get_slot_number_by_color(self, color):
+        """
+        This function is to find a list of slot number based on car color
+        """
+        slot_list = []
+        if self.car_count == 0:
+            return 'No car in parking lot!'
+        for slot in self.parking_lot_list:
+            if slot.car is None:
+                continue
+            else:
+                if slot.get_color() == color:
+                    slot_list.append(slot.id)
+        if len(slot_list) == 0:
+            return 'No car with this color!'
+        else:
+            id_with_color = ' '.join([str(id) for id in slot_list])
+            print(f'Id list:{id_with_color}')
+            return id_with_color
+
+    def get_slot_number_by_registration_number(self, registration_number):
+        """
+        This function is to get the slot number based on car registration number
+        """
+        if self.car_count == 0:
+            return 'No car in parking lot!'
+        for slot in self.parking_lot_list:
+            if slot.car is None:
+                continue
+            else:
+                if slot.get_registration_number() == registration_number:
+                    print(f'Id: {slot.id}')
+                    return slot.id
+        return 'Car not found!'
+    
+    def get_registration_numbers_with_colour(self, color):
+        """
+        This function is to get the registration number list based on car color
+        """
+        reg_num_list = []
+        if self.car_count == 0:
+            return 'No car in parking lot!'
+        for slot in self.parking_lot_list:
+            if slot.car is None:
+                continue
+            else:
+                if slot.get_color() == color:
+                    reg_num_list.append(slot.get_registration_number())
```

Comment:
instead of adding the registration number of the car like this, could I do something like `slots += slot`, and later I do `slots.get_registration_numer()  # [reg1, reg2, reg3]`. Could you make this possible?